packages/core/src/messages/responses/cache-ttl-decrease.ts

```diff
@@ -58,4 +42,13 @@ class _Error extends Response {
  * - `message()` - a human-readable description of the error
  * - `innerException()` - the original error that caused the failure; can be re-thrown.
  */
-export class Error extends ResponseError(_Error) {}
+export class Error extends BaseResponseError implements IResponse {
+  constructor(_innerException: SdkError) {
+    super(_innerException);
+  }
+
+  readonly type: CacheDecreaseTtlResponse.Error =
+    CacheDecreaseTtlResponse.Error;
+}
+
+export type Response = Set | NotSet | Miss | Error;
```

Comment:
I saw that JS and Go implemented only Set/Miss/Error for the update/increase/decrease ttl responses, but in Rust we implemented NotSet as well because it was possible a precondition for increase/decrease ttl would fail ([Rust int test for reference](https://github.com/momentohq/client-sdk-rust/blob/main/sdk/tests/cache/ttl.rs#L198)), but not for update ttl.

Should we audit the SDKs for consistency on this front? Could be:

- Set/NotSet/Miss/Error for all 3 commands (update, increase, decrease ttl)
- Set/NotSet/Miss/Error for increase and decrease, but Set/Miss/Error for update
- Set/Miss/Error for all 3 commands

Reply:
good call I would probably try and align as much as possible with the protos

Reply:
thinking about this, will reply with more thoughts soon

Reply:
I definitely think we should include `NotSet` in the enums where it is a possible response type; let's eliminate the `FailedPrecondition` option.

I'm on the fence about whether to include `NotSet` for the cases where it is not actually a possible return type. I think I would vote for *not* including it on those. We can always add it in the future, but technically we can't remove it.

Reply:
and yes we should create tickets for adding `NotSet` in any of the SDKs where it got skipped.

Reply:
i created a generic ticket for the missing `NotSet` in other SDKs and put it on the board